src/structures/MessageAttachment.js

```diff
@@ -43,6 +43,20 @@ class MessageAttachment {
     return this;
   }
 
+  /**
+   * Sets whether this attachment is a spoiler
+   * @param {boolean} spoiler Whether the attachment should be marked as a spoiler
+   * @returns {MessageAttachment} This attachment
+   */
+  setSpoiler(spoiler) {
+    if (spoiler === true && !this.name.startsWith('SPOILER_')) {
+      this.name = `SPOILER_${this.name}`;
+      return this;
+    }
+    if (this.name.startsWith('SPOILER_')) this.name.replace('SPOILER_', '');
```

Comment:
What if spoiler is true and the name already starts with SPOILER_ ? I believe this should also be checked for if spoiler is false
Also spoiler should probably be true by default like other setX methods